### ISSUE TITLE
einfo applet printf format handling

### ISSUE THREAD

**Issue by TerraTech** [OpenRC/openrc#260]
Gentoo: default/linux/amd64/17.0/hardened (stable)
OpenRC: sys-apps/openrc-0.38.3

Looking at the einfo.3 manpage, I see that the library supports printf format handling, however the `src/rc/do_e.c` applet handler doesn't seem to allow for this:  Neither by direct binary execution nor by sourcing in /lib/rc/sh/functions.sh

```
int
einfo(const char * restrict format, ...);
```

```sh
$ /lib/rc/bin/einfo "%s" "test"
 * %s test
```
Was it intentional when the applet executor was designed that it not allow printf format handling?

As it stands, the only way to work around this is to perform a subshell fork to printf, or a back-to-back call:
```sh
 $ /lib/rc/bin/einfo $(printf "%s" "test")
 * test
-OR-
$ /lib/rc/bin/einfon ""
$ printf "%s" "test"   (*to avoid the subshell fork)
 * test
```

TIA!

**Comment by vapier** [comment]
afaict, the e* programs have behaved this way for over a decade.  based on the lack of complaints, prob best to just leave it that way.

**Comment by TerraTech** [comment]
Yes, but adding printf formatting ability doesn't take anything away.  It is an enhancement that aligns to the documented behavior with 100% backwards compatibility.

**Comment by vapier** [comment]
maybe you can convince William. I don't expect the code to be that complicated to pull off, it's just that at this point, tools have migrated away from these low level layers to the point where the only realistic user is openrc itself. and it doesn't seem like this feature would be utilized there.
